pep/app/pods/most-cited/controller.ts

```diff
@@ -7,30 +7,48 @@ import { inject as service } from '@ember/service';
 import LoadingBarService from 'pep/services/loading-bar';
 import FastbootMediaService from 'pep/services/fastboot-media';
 import SidebarService from 'pep/services/sidebar';
-
-type PossiblePeriodValues = '5' | '10' | '20' | 'all';
-type period = { label: string; value: PossiblePeriodValues };
+import IntlService from 'ember-intl/services/intl';
+import Journal from 'pep/pods/journal/model';
+import { PERIODS, PossiblePeriodValues } from 'pep/constants/sidebar';
 
 export default class MostCited extends Controller {
     @service loadingBar!: LoadingBarService;
     @service fastbootMedia!: FastbootMediaService;
     @service sidebar!: SidebarService;
+    @service intl!: IntlService;
 
     queryParams = ['author', 'title', 'sourcename', 'period'];
```

Comment:
i think we should look at this later in another PR, but i think its not the best UX that as the user types in filter fields, the URL query params update immediately. There is the browser history pollution issue (tho that can be partly fixed with `replace: true` configs), but also, i think its slightly nicer to not update the query params until a new search has been run w/the new filter values (basically the behavior of the current Search page)

Reply:
Agreed completely